contracts/Burner.cdc

```diff
@@ -0,0 +1,79 @@
+// Burner is a contract that can facilitate the destruction of any resource on flow.
+//
+// Contributors
+// - Austin Kline - https://twitter.com/austin_flowty
+// - Deniz Edincik - https://twitter.com/bluesign
+// - Bastian Müller - https://twitter.com/turbolent
+pub contract Burner {
+    // When Crescendo (Cadence 1.0) is released, custom destructors will be removed from cadece.
+    // Burnable is an interface meant to replace this lost feature, allowing anyone to add a callback
+    // method to ensure they do not destroy something which is not meant to be, or to add logic based on destruction
+    // such as tracking the supply of an NFT Collection
+    //
+    // NOTE: The only way to see benefit from this interface is to call the burnCallback method yourself,
+    // or to always use the burn method in this contract. Anyone who owns a resource can always elect **not**
+    // to destroy a resource this way
+    pub resource interface Burnable {
+        pub fun burnCallback()
```

Comment:
```suggestion
        access(contract) fun burnCallback()
```

If this method is public and we use it to emit the `FungibleToken.Burned` event, wouldn't that be able to be spoofed or spammed since anyone can call it from their resource? I propose making this `access(contract)` so that only the `burn` function in this contract can call it.

Reply:
wait, is that even possible? If I make an `access(contract)` method in an interface, does that mean that only the contract the defines the resource interface implementation can access it, or does it mean the `Burner` contract can access it?